### DISCUSSION TITLE
Set ID scalar to Int when setting db idField kind to 'autoincrement'

### DISCUSSION THREAD

**Discussion by oldo** [keystonejs/keystone#7042] [Feature requests]
I have set the database id field to `autoincrement`

```
import { config } from '@keystone-6/core';
import { session, withAuth } from './auth';
import extendGraphqlSchema from './schema/extensions';
import lists from './schema/lists';

export default withAuth(
  config({
    db: {
      provider: 'sqlite',
      url: 'file:./keystone.db',
      idField: { kind: 'autoincrement' }, <-------------
    },
    extendGraphqlSchema,
    ui: { isAccessAllowed: context => !!context.session?.data },
    lists,
    session,
  })
);
```

However the ID scalar is still set as a `String`. Is there any way to redefine it as `Int`? It would just avoid me having to do `id.toString()` in all my custom resolvers - which is not a big deal but nice to have typings/schema actually reflect how the db is setup.

**Comment by dcousens** [comment]
Related to https://github.com/keystonejs/keystone/pull/6048 ?

**Reply by oldo** [reply]
Thanks @dcousens 
Yeah, it is related, although not quite as broad in scope as what is being proposed there. The only functionality I'm after is to be able to override the id scalar type.
Looks like this functionality hasn't been incorporated into the main release (yet?). Fingers crossed we get the option as it'll make the system a bit more type-complete.
Cheers.
